### ISSUE TITLE
Question: how to add new nodes to JADE

### ISSUE THREAD

**Issue by odow** [EPOC-NZ/JADE.jl#45]
Zainab asks:

_I am using JADE which is one of the applications of the JuMP package. Ther results of my simulations vary too much. I was trying to add another node to the EMI version of JADE. Any help in this matter would be much appreciated._

x-ref: https://discourse.julialang.org/t/adding-nodes-to-jade/106774

**Comment by odow** [comment]
The answer is that you need to go through _all_ of the data files, and anywhere that you see `HAY`, `NI`, or `SI`, you need to add new data for your new nodes.

For example:
 * `demand.csv`: You'll need new demand data for each node
 * `lost_load.csv`: you'll need new load-shedding data for the node
 * `transmission.csv`: you'll need new transmission data for the network
 * `transmission_outages.csv`: you'll need new transmission outages for the network

You shouldn't need to modify any of the Julia code, only the input data files.

**Comment by ZBRizvi** [comment]
Thank you for your answer.
I did go through the files you mentioned. Quick question: Do we need to update the historical data files too?

**Comment by odow** [comment]
Yes, if you're going to train using historical data, then you'll need that. I don't know if I can prescribe exactly every place that needs updating, but if something broke, it probably means you're missing something from a data file. I wouldn't try changing the code to fix things.

@adow031 and @DrAndyPhilpott are the real experts here though. I have a very limited understanding of JADE.

**Comment by ZBRizvi** [comment]
Thanks Oscar :-) 
I was wondering if information about any new node needs to be added to the inflows.csv file?

**Comment by odow** [comment]
The inflows are for the hydro reservoirs. If you start adding new nodes, you might want to change the node of some lakes like `Lake_Waikaremoana`.

Stepping back, I think some questions to think about are: what are you trying to model, why are you trying to model it, what is the output that you expect, and what level of fidelity is necessary to get actionable outputs?

Let's skip over the node question for now. Assume you have a perfect graph of the network. How do you plan to model the microgrid? What temporal scale? What are the outputs?

The main goal of JADE is a medium-term valuation of water in the southern lakes. I worry slightly that a small change in an isolated part of the country will have a very minor impact on that valuation. If you want to say, look at nodal power prices 30 minute blocks over a year to see if consumers in Gisborne pay more or less, I don't think JADE can answer that question for you. You might want to take a look at https://www.emi.ea.govt.nz/Wholesale/Tools/vSPD. I'd have a chat with Mike and Andy to discuss.

**Comment by ZBRizvi** [comment]
I agree about discussing with Mike and Andy. I think they did have a discussion before we started working in JADE. But it would be a good idea to explore vSPD again.
Thanks for you input Oscar. Will look into vSPD more.
